client/state/jetpack-connect/actions.js

```diff
@@ -196,32 +196,72 @@ export function retryAuth( url, attemptNumber ) {
 	};
 }
 
-export function createAccount( userData ) {
+export function createAccount( userData, { service, access_token, id_token } = {} ) {
```

Comment:
A lot of changes here; maybe it could benefit from some tests.

Reply:
👍 True.